api/src/main/java/ai/djl/engine/Engine.java

```diff
@@ -302,6 +307,11 @@ public SymbolBlock newSymbolBlock(NDManager manager) {
      */
     public abstract NDManager newBaseManager(Device device);
 
+    public LMBlock newLMAdapter(String languageModel, GPTConfig gptConfig, Block[] blocks)
```

Comment:
It doesn't need to be engine agnostic. It should be somewhere in `pytorch-model-zoo`. In the `PtModelZoo`, register the GPT2PtLMBlock model. Within the metadata for that model, you can specify the block factory to use. Maybe a nested class like `GPT2PtLMBlock.Factory`. And, that block factory (probably located in `pytorch-model-zoo`) will call the `GPT2PtLMBlock(...)` constructor. Hopefully that makes sense

Reply:
It could be kept in the same directory, but would probably be a bit better in the `pytorch-model-zoo` package rather than `pytorch-engines` package. Within the `pytorch-model-zoo`, it still depends with implementation on the `pytorch-engines` so you can still access all of the internal pytorch details